src/main/java/htsjdk/samtools/cram/ref/ReferenceContext.java

```diff
@@ -0,0 +1,152 @@
+package htsjdk.samtools.cram.ref;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.cram.CRAMException;
+
+/**
+ * Represents the current state of reference sequence processing.
+ *
+ * Are we handling MULTI_REFERENCE records (-2, from the CRAM spec)?
+ *
+ * Are we handling UNMAPPED_UNPLACED records (-1, from the CRAM spec)?
+ *
+ * Or are we handing a known SINGLE_REFERENCE sequence (0 or higher, from the CRAM spec))?
+ *
+ */
+public class ReferenceContext implements Comparable<ReferenceContext> {
+    public static final int REF_ID_MULTIPLE = -2;
+    public static final int REF_ID_UNMAPPED = SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX; // -1
+
+    public static final ReferenceContext MULTIPLE = new ReferenceContext(REF_ID_MULTIPLE);
+    public static final ReferenceContext UNMAPPED = new ReferenceContext(REF_ID_UNMAPPED);
```

Comment:
Suggest aligning the names for the integer constants, these static objects, and the enum names as much as possible. For example rather than `REF_ID_MULTIPLE`, `MULTIPLE` , and `MULTI_REFERENCE`, which use differing abbreviations and ordering of ref/multi, maybe use `REFERENCE_MULTIPLE_ID`, `REFERENCE_MULTIPLE`, and `REFERENCE_MULTIPLE`. Same then the same for corresponding `SINGLE` and `UNMAPPED` ones, etc. Or whatever makes sense.